src/components/webcams/WebrtcRTSPSimpleServer.vue

```diff
@@ -0,0 +1,251 @@
+
+    @Ref()
+    declare video: HTMLVideoElement
+
+    // webrtc player vars
+    private webRtcTerminated: boolean = false
+    private webRtcWs: any
+    private webRtcPc: any
+    private webRtcRestartTimeout: any
+
+    mounted() {
+        this.video.onload = () => {
+            if (this.webRtcTerminated) {
+                this.terminateWebRtcVideo() // catch a potential race condition
+            } else {
+                this.video.play() // trigger autoplay for some browsers
+            }
+        }
+    }
+
+    get webcamStyle() {
+        let transforms = ''
+        if ('flipX' in this.camSettings && this.camSettings.flipX) transforms += ' scaleX(-1)'
+        if ('flipX' in this.camSettings && this.camSettings.flipY) transforms += ' scaleY(-1)'
+        if (transforms.trimLeft().length) return { transform: transforms.trimLeft() }
+
+        return ''
+    }
+
+    isWebRtcUrl(): boolean {
+        return this.camSettings.urlStream.startsWith('ws://') || this.camSettings.urlStream.startsWith('wss://')
+    }
+
+    terminateWebRtcVideo() {
+        try {
+            this.video.pause()
+        } catch (err) {
+            // ignore - more important to shut down the sockets.
+        }
+
+        this.webRtcTerminate()
+    }
+
+    visibilityChanged(isVisible: boolean) {
+        this.isVisible = isVisible
+
+        if (!this.isVisible) {
+            this.terminateWebRtcVideo()
+        } else {
+            this.webRtcStart()
+            // // TODO: better way to detect ready to play?
+            // setTimeout(() => {
+            //     this.video.play()
+            // }, 500)
+        }
+    }
+
+    beforeDestroy() {
+        this.terminateWebRtcVideo()
+    }
+
+    // webrtc player methods
+    // adapated from sample player in https://github.com/mrlt8/docker-wyze-bridge
+    webRtcStart() {
+        // unterminate.. we're starting again.
+        this.webRtcTerminated = false
+
+        console.log('connecting')
+
+        if (!this.isWebRtcUrl()) {
+            console.error('not a webRtcSrc url')
+            return
+        }
+
+        this.webRtcWs = new WebSocket(this.camSettings.urlStream)
+
+        this.webRtcWs.onerror = () => {
+            console.log('ws error')
+            if (this.webRtcWs === null) {
+                return
+            }
+            this.webRtcWs.close()
+            this.webRtcWs = null
+        }
+
+        this.webRtcWs.onclose = () => {
+            console.log('ws closed')
+            this.webRtcWs = null
+            this.webRtcScheduleRestart()
+        }
+
+        this.webRtcWs.onmessage = (msg: any) => this.webRtcOnIceServers(msg)
+    }
+
+    webRtcTerminate() {
+        this.webRtcTerminated = true
+
+        if (this.webRtcWs) {
+            try {
+                this.webRtcWs.close()
+            } catch (err) {
+                // ignore
+            }
+        }
+        if (this.webRtcPc) {
+            try {
+                this.webRtcPc.close()
```

Comment:
this.webRtcPs?.close() should be enough here. i don't think you need a try...catch statement